flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/history/HistoryServerArchiveFetcher.java

```diff
@@ -153,9 +157,20 @@ public void run() {
 						continue;
 					}
 					boolean updateOverview = false;
+					long now = System.currentTimeMillis();
 					for (FileStatus jobArchive : jobArchives) {
 						Path jobArchivePath = jobArchive.getPath();
 						String jobID = jobArchivePath.getName();
+						if (retainedApplicationsMillis > 0L && now - jobArchive.getModificationTime() > retainedApplicationsMillis) {
+							if (LOG.isDebugEnabled()) {
+								LOG.debug("delete the old archived job for path {}." + jobArchivePath.toString());
+							}
+							jobArchivePath.getFileSystem().delete(jobArchivePath, false);
+							continue;
+						} else {
+							LOG.warn("Negative or zero values {} of the historyserver.archive.fs.retained-application-millis " +
```

Comment:
Maybe we will come into the else branch when `retainedApplicationsMillis` > 0?

Reply:
I am very sorry that my carelessness caused this mistake, Thanks a lot for reviewing it again!